src/blenderbim/blenderbim/bim/module/georeference/operator.py

```diff
@@ -17,345 +17,146 @@
-            return True
-
-
-class SetBlenderGridNorth(bpy.types.Operator):
-    bl_idname = "bim.set_blender_grid_north"
-    bl_label = "Set Blender Grid North"
+        core.edit_georeferencing(tool.Ifc, tool.Georeference)
+    
+class DisableEditingGeoreferencing(bpy.types.Operator, tool.Ifc.Operator):
+    bl_idname = "bim.disable_editing_georeferencing"
+    bl_label = "Disable Editing Georeferencing"
     bl_options = {"REGISTER", "UNDO"}
+    bl_description = "Close editing panel"
 
-    def execute(self, context):
-        context.scene.sun_pos_properties.north_offset = -radians(
-            ifcopenshell.util.geolocation.xaxis2angle(
-                float(context.scene.BIMGeoreferenceProperties.map_conversion.get("XAxisAbscissa").string_value),
-                float(context.scene.BIMGeoreferenceProperties.map_conversion.get("XAxisOrdinate").string_value),
-            )
-        )
-        return {"FINISHED"}
-
+    def _execute(self, context):
+        core.disable_editing_georeferencing(tool.Georeference)
 
-class SetIfcGridNorth(bpy.types.Operator):
+class SetIfcGridNorth(bpy.types.Operator, tool.Ifc.Operator):
     bl_idname = "bim.set_ifc_grid_north"
     bl_label = "Set IFC Grid North"
     bl_options = {"REGISTER", "UNDO"}
+    bl_description = "Set IFC grid north"
+    
+    def _execute(self, context):
+        core.set_ifc_grid_north()
 
-    def execute(self, context):
-        x_angle = -context.scene.sun_pos_properties.north_offset
-        context.scene.BIMGeoreferenceProperties.map_conversion.get("XAxisAbscissa").string_value = str(cos(x_angle))
-        context.scene.BIMGeoreferenceProperties.map_conversion.get("XAxisOrdinate").string_value = str(sin(x_angle))
-        return {"FINISHED"}
-
-
-class SetBlenderTrueNorth(bpy.types.Operator):
-    bl_idname = "bim.set_blender_true_north"
-    bl_label = "Set Blender True North"
+class SetBlenderGridNorth(bpy.types.Operator, tool.Ifc.Operator):
+    bl_idname = "bim.set_blender_grid_north"
+    bl_label = "Set Blender Grid North"
     bl_options = {"REGISTER", "UNDO"}
+    bl_description = "Set Blender grid north"
+    
+    def _execute(self, context):
+        core.set_blender_grid_north()
 
-    def execute(self, context):
-        context.scene.sun_pos_properties.north_offset = -radians(
-            ifcopenshell.util.geolocation.yaxis2angle(
-                float(context.scene.BIMGeoreferenceProperties.true_north_abscissa),
-                float(context.scene.BIMGeoreferenceProperties.true_north_ordinate),
-            )
-        )
-        return {"FINISHED"}
-
+class GetCursorLocation(bpy.types.Operator, tool.Ifc.Operator):
+    bl_idname = "bim.get_cursor_location"
+    bl_label = "Get Cursor Location"
+    bl_options = {"REGISTER", "UNDO"}
+    bl_description = "Insert the current cursor coordinates"
+    
+    @classmethod
+    def poll(cls, context): #TODO is it right to use the poll method?
```

Comment:
Good point, let's simplify this to just polling return tool.Ifc.get()

Reply:
So the following is ok?

@classmethod
def poll(cls, context):
    return file = tool.Ifc.get()

Reply:
Yep

Reply:
ok, l'll fix.